graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/FunctionBodyGenerator.java

```diff
@@ -173,18 +177,72 @@ public FunctionBodyNode generateInitFunctionBody(AuthConfig authConfig) {
      * @param authConfig the object instance representing authentication configuration information
      * @return           the object instance representing http client config
      */
-    private StatementNode createHttpClientConfig(AuthConfig authConfig) {
-        String auth = "";
+    private List<StatementNode> generateHttpClientConfigurationNode(AuthConfig authConfig) {
+        List<StatementNode> assignmentNodes = new ArrayList<>();
+        NodeList<AnnotationNode> annotationNodes = createEmptyNodeList();
+
+        // GraphQL {@code variables} declaration
+        BuiltinSimpleNameReferenceNode typeBindingPattern = createBuiltinSimpleNameReferenceNode(null,
+                createIdentifierToken(HTTP_CLIENT_CONFIG_TYPE_NAME));
+        CaptureBindingPatternNode bindingPattern = createCaptureBindingPatternNode(
+                createIdentifierToken("httpClientConfig"));
+        TypedBindingPatternNode typedBindingPatternNode = createTypedBindingPatternNode(typeBindingPattern,
+                bindingPattern);
+
+        // Expression node
+        List<Node> defaultFields = new ArrayList<>();
+
+        int count = 0;
+        Map<String, String> map = new LinkedHashMap<>();
         if (authConfig.isClientConfig()) {
-            auth = "auth: config.auth,";
+            map.put("auth", "config.auth");
+        }
+        map.put("httpVersion", "config.httpVersion");
+        map.put("timeout", "config.timeout");
+        map.put("forwarded", "config.forwarded");
+        map.put("poolConfig", "config.poolConfig");
+        map.put("compression", "config.compression");
+        map.put("circuitBreaker", "config.circuitBreaker");
+        map.put("retryConfig", "config.retryConfig");
+
+        for (Map.Entry entry : map.entrySet()) {
+            BuiltinSimpleNameReferenceNode valueExpr = createBuiltinSimpleNameReferenceNode(null,
+                    createIdentifierToken((entry.getValue().toString())));
+            SpecificFieldNode specificFieldNode = createSpecificFieldNode(null,
+                    createIdentifierToken(entry.getKey().toString()), createToken(COLON_TOKEN), valueExpr);
+            defaultFields.add(specificFieldNode);
+            count++;
+            if (count < map.size()) {
+                defaultFields.add(createToken(COMMA_TOKEN));
+            }
+        }
+        MappingConstructorExpressionNode initializer = createMappingConstructorExpressionNode(
+                createToken(OPEN_BRACE_TOKEN), createSeparatedNodeList(defaultFields), createToken(CLOSE_BRACE_TOKEN));
+
+        assignmentNodes.add(createVariableDeclarationNode(annotationNodes, null, typedBindingPatternNode,
+                createToken(EQUAL_TOKEN), initializer, createToken(SEMICOLON_TOKEN)));
+        generateOptionalFields(assignmentNodes);
+        return assignmentNodes;
+    }
+
+    private void generateOptionalFields(List<StatementNode> optionalFields) {
```

Comment:
Let's add some documentation comments here for readability.

Reply:
Added in https://github.com/ballerina-platform/graphql-tools/commit/820d69eccf3d162405d31e4797434fb404c1da4e